packages/sakuli-legacy/src/context/common/region/sakuli-region.class.spec.ts

```diff
@@ -36,142 +43,146 @@ describe("sakuli region class", () => {
-  });
+      //THEN
+      expect(runAsActionSpy).toBeCalledWith(
+        testExecutionContextMock,
+        "rightClick",
+        expect.any(Function)
+      );
+      expect(mouseApi.move).toBeCalledWith(sakuliRegion);
+      expect(mouseApi.rightClick).toBeCalled();
+      expect(region).toBeInstanceOf(sakuliRegionClass);
+    });
 
-  it("should scroll down on mouseWheelDown", async () => {
-    //GIVEN
-    const numberOfSteps = 42;
-
-    //WHEN
-    const region = await sakuliRegion.mouseWheelDown(numberOfSteps);
-
-    //THEN
-    expect(runAsActionSpy).toBeCalledWith(
-      testExecutionContextMock,
-      "mouseWheelDown",
-      expect.any(Function)
-    );
-    expect(mouseApi.scrollDown).toBeCalledWith(numberOfSteps);
-    expect(region).toBeInstanceOf(sakuliRegionClass);
-  });
+    it("should press given mouse button on mouseDown", async () => {
+      //GIVEN
+      const button = MouseButton.MIDDLE;
 
-  it("should scroll up on mouseWheelUp", async () => {
-    //GIVEN
-    const numberOfSteps = 42;
-
-    //WHEN
-    const region = await sakuliRegion.mouseWheelUp(numberOfSteps);
-
-    //THEN
-    expect(runAsActionSpy).toBeCalledWith(
-      testExecutionContextMock,
-      "mouseWheelUp",
-      expect.any(Function)
-    );
-    expect(mouseApi.scrollUp).toBeCalledWith(numberOfSteps);
-    expect(region).toBeInstanceOf(sakuliRegionClass);
-  });
+      //WHEN
+      const region = await sakuliRegion.mouseDown(button);
+
+      //THEN
+      expect(runAsActionSpy).toBeCalledWith(
+        testExecutionContextMock,
+        "mouseDown",
+        expect.any(Function)
+      );
+      expect(mouseApi.pressButton).toBeCalledWith(button);
+      expect(registerMouseDown).toBeCalledWith(button);
+      expect(region).toBeInstanceOf(sakuliRegionClass);
+    });
 
-  it("should move to specified destination", async () => {
-    //GIVEN
-    const destination = new sakuliRegionClass(42, 84, 21, 168);
+    it("should release given mouse button on mouseUp", async () => {
+      //GIVEN
+      const button = MouseButton.LEFT;
 
-    //WHEN
-    await sakuliRegion.moveTo(destination);
+      //WHEN
+      const region = await sakuliRegion.mouseUp(button);
 
-    //THEN
-    expect(mouseApi.move).toBeCalledWith(destination);
+      //THEN
+      expect(runAsActionSpy).toBeCalledWith(
+        testExecutionContextMock,
+        "mouseUp",
+        expect.any(Function)
+      );
+      expect(mouseApi.releaseButton).toBeCalledWith(button);
+      expect(registerMouseUp).toBeCalledWith(button);
+      expect(region).toBeInstanceOf(sakuliRegionClass);
+    });
+
+    it("should drag and drop to target region", async () => {
+      //WHEN
+      const region = await sakuliRegion.dragAndDropTo(sakuliRegion);
+
+      //THEN
+      expect(runAsActionSpy).toBeCalledWith(
+        testExecutionContextMock,
+        "dragAndDropTo",
+        expect.any(Function)
+      );
+      expect(mouseApi.dragAndDrop).toBeCalledWith(sakuliRegion);
```

Comment:
A drag & drop action presses and releases the left mousebutton before / after moving.
We should register these two event as well.

Reply:
Thx! :+1:

Reply:
As far as I can see, this is an upstream functionality provided by nut.js. Sakuli is just using `await mouse.drag(straightTo(centerOf(await toNutRegion(region))));` from nut.

Reply:
https://github.com/nut-tree/nut.js/blob/28f2013982378335691b0d5a73c2c78ca83f2773/lib/mouse.class.ts#L178-L190